source/common/config/utility.h

```diff
@@ -310,7 +311,14 @@ class Utility {
   template <class Factory, class ProtoMessage>
   static Factory* getAndCheckFactory(const ProtoMessage& message, bool is_optional) {
     Factory* factory = Utility::getFactoryByType<Factory>(message.typed_config());
-    if (factory != nullptr) {
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.prefer_extension_type_url")) {
```

Comment:
Maybe we should call the runtime flag as `using__extension_type_url`.  `prefer` sounds like using the type URL first, then fallback to the name, but actually it isn't.